### PR TITLE
Improved the Instructions

### PR DESCRIPTION
**Pull request by AlexejPenner**
Recently some problems in the computer vision project have come to light with missing instructions on the label studio setup. 

Additionally, I identified some problems with the pip installation and the requirement for a Huggingface account.

### CODE REVIEW

**Review by strickvl — CHANGES_REQUESTED** [review]
You should check the latest (i.e. post pydantic v2) changes to label studio. They're incompatible with these new instructions. We ask people to use docker-compose to spin up their LS instance now, and the installation is of the SDK package for label studio. So I think this needs aligning with the new way, unless ZenML is pinned to a pre-Pydantic v2 version (in which case, this example probably wants updating IMO).

**Review by strickvl — CHANGES_REQUESTED** [review]
See https://github.com/zenml-io/zenml-projects/pull/113#pullrequestreview-2138287978

File: end-to-end-computer-vision/README.md [hunk 1]
```diff
@@ -32,7 +32,8 @@ deployed instance of zenml:
 
 ```bash
 pip install -r requirements.txt
-zenml integration install label_studio torch gcp mlflow -y
+zenml integration install torch gcp mlflow -y
+pip install label-studio
```
Comment (strickvl) [inline end-to-end-computer-vision/README.md:36]:
```suggestion
```

I think this is no longer needed? And maybe you can add label_studio integration here too instead of below?

File: end-to-end-computer-vision/README.md [hunk 2]
```diff
@@ -63,37 +64,52 @@ zenml connect --url <INSERT_ZENML_URL_HERE>
...
-[Learn how to set up a Vertex step operator stack component within zenml
-here](https://docs.zenml.io/stack-components/step-operators/vertex)
-### Set up Container Registry
-
-[Learn how to set up a google cloud container registry component within zenml
-here](https://docs.zenml.io/stack-components/container-registries/gcp)
+3) Set up a bucket to use as artifact store within ZenML
+Learn how to set up a GCP artifact store stack component within zenml
+[here](https://docs.zenml.io/stack-components/artifact-stores)
+4) Set up vertex for pipeline orchestration
```
Comment (strickvl) [inline end-to-end-computer-vision/README.md:74]:
```suggestion
4) Set up Vertex for pipeline orchestration
```

File: end-to-end-computer-vision/README.md [hunk 3]
```diff
@@ -63,37 +64,52 @@ zenml connect --url <INSERT_ZENML_URL_HERE>
...
+Learn how to set up a Vertex orchestrator stack component within zenml
+[here](https://docs.zenml.io/stack-components/orchestrators/vertex)
+5) For training on accelerators like GPUs/TPUs set up Vertex
+Learn how to set up a Vertex step operator stack component within zenml
+[here](https://docs.zenml.io/stack-components/step-operators/vertex)
+6) Set up Container Registry in GCP. Learn how to set up a google cloud container registry component within zenml
```
Comment (strickvl) [inline end-to-end-computer-vision/README.md:80]:
```suggestion
6) Set up a Container Registry in GCP. Learn how to set up a google cloud container registry component within ZenML
```

File: end-to-end-computer-vision/README.md [hunk 4]
```diff
@@ -63,37 +64,52 @@ zenml connect --url <INSERT_ZENML_URL_HERE>
...
-### For training on accelerators like GPUs/TPUs set up Vertex
-
-[Learn how to set up a Vertex step operator stack component within zenml
-here](https://docs.zenml.io/stack-components/step-operators/vertex)
-### Set up Container Registry
-
-[Learn how to set up a google cloud container registry component within zenml
-here](https://docs.zenml.io/stack-components/container-registries/gcp)
+3) Set up a bucket to use as artifact store within ZenML
+Learn how to set up a GCP artifact store stack component within zenml
```
Comment (strickvl) [inline end-to-end-computer-vision/README.md:72]:
```suggestion
3) Set up a bucket to use as artifact store within ZenML
Learn how to set up a GCP artifact store stack component within ZenML
```

File: end-to-end-computer-vision/README.md (hunk 3 above)
Comment (strickvl) [inline end-to-end-computer-vision/README.md:80]:
```suggestion
Learn how to set up a Vertex orchestrator stack component within ZenML
[here](https://docs.zenml.io/stack-components/orchestrators/vertex)
5) For training on accelerators like GPUs/TPUs set up Vertex
Learn how to set up a Vertex step operator stack component within zenml
[here](https://docs.zenml.io/stack-components/step-operators/vertex)
6) Set up Container Registry in GCP. Learn how to set up a google cloud container registry component within ZenML
```

File: end-to-end-computer-vision/README.md [hunk 5]
```diff
@@ -63,37 +64,52 @@ zenml connect --url <INSERT_ZENML_URL_HERE>
...
 
-### [Start Label Studio locally](https://labelstud.io/guide/start)
-### [Follow these ZenML instructions to set up Label Studio as a stack component](https://docs.zenml.io/stack-components/annotators/label-studio)
-### Create a project within Label Studio and name it `ship_detection_gcp`
-### [Set up Label Studio to use external storage](https://labelstud.io/guide/storage) 
-use the first bucket that you created to data persistence
+1) [Start Label Studio locally](https://labelstud.io/guide/start)
+For label studio we recommend using docker/docker-compose to deploy a local instance
+```bash
+zenml integration install label_studio
```
Comment (strickvl) [inline end-to-end-computer-vision/README.md:88]:
```suggestion
```

No need for this here if you did it above already.

File: end-to-end-computer-vision/README.md [hunk 6]
```diff
@@ -63,37 +64,52 @@ zenml connect --url <INSERT_ZENML_URL_HERE>
...
 
 ## Label Studio
 
-### [Start Label Studio locally](https://labelstud.io/guide/start)
-### [Follow these ZenML instructions to set up Label Studio as a stack component](https://docs.zenml.io/stack-components/annotators/label-studio)
-### Create a project within Label Studio and name it `ship_detection_gcp`
-### [Set up Label Studio to use external storage](https://labelstud.io/guide/storage) 
-use the first bucket that you created to data persistence
+1) [Start Label Studio locally](https://labelstud.io/guide/start)
+For label studio we recommend using docker/docker-compose to deploy a local instance
```
Comment (strickvl) [inline end-to-end-computer-vision/README.md:86]:
```suggestion
For Label Studio we recommend using docker/docker-compose to deploy a local instance
```

File: end-to-end-computer-vision/README.md [hunk 7]
```diff
@@ -63,37 +64,52 @@ zenml connect --url <INSERT_ZENML_URL_HERE>
...
+2) [Follow these ZenML instructions to set up Label Studio as a stack component](https://docs.zenml.io/stack-components/annotators/label-studio#how-to-deploy-it)
+3) Create a project within Label Studio and name it `ship_detection_gcp`
+![img.png](_assets/project_creation_label_studio.png)
+4) Configure your project to use `Object Detection with Bounding Boxes` as Labeling Setup
+![img.png](_assets/labeling_setup.png)
+In the following screen you now need to configure the labeling interface. This is where you define the different classes that you want to detect. In our case this should be a single `ship` class.
+![img.png](_assets/labeling_interface.png)
+Additionally you might want to allow users to zoom during labeling. This can be configured when you scroll down on this same screen.
+6) [Set up Label Studio to use external storage](https://labelstud.io/guide/storage) 
+use the first bucket that you created for data persistence
```
Comment (strickvl) [inline end-to-end-computer-vision/README.md:103]:
```suggestion
Use the first bucket that you created for data persistence
```

File: end-to-end-computer-vision/README.md [hunk 8]
```diff
@@ -63,37 +64,52 @@ zenml connect --url <INSERT_ZENML_URL_HERE>
...
+
+## Huggingface
+
+This specific project relies on a dataset loaded from huggingface. As such a free huggingface account is needed.
```
Comment (strickvl) [inline end-to-end-computer-vision/README.md:107]:
```suggestion
This specific project relies on a dataset loaded from Hugging Face. As such a free Hugging Face account is needed.
```

File: end-to-end-computer-vision/README.md [hunk 9]
```diff
@@ -63,37 +64,52 @@ zenml connect --url <INSERT_ZENML_URL_HERE>
...
+![img.png](_assets/project_creation_label_studio.png)
+4) Configure your project to use `Object Detection with Bounding Boxes` as Labeling Setup
+![img.png](_assets/labeling_setup.png)
+In the following screen you now need to configure the labeling interface. This is where you define the different classes that you want to detect. In our case this should be a single `ship` class.
+![img.png](_assets/labeling_interface.png)
+Additionally you might want to allow users to zoom during labeling. This can be configured when you scroll down on this same screen.
+6) [Set up Label Studio to use external storage](https://labelstud.io/guide/storage) 
+use the first bucket that you created for data persistence
+
+## Huggingface
```
Comment (strickvl) [inline end-to-end-computer-vision/README.md:105]:
```suggestion
## Hugging Face
```

File: end-to-end-computer-vision/configs/ingest_data.yaml [hunk 10]
```diff
@@ -5,7 +5,7 @@ steps:
     enable_step_logs: False
     parameters:
       dataset: "datadrivenscience/ship-detection"
-      data_source: # Insert your bucket path here where the training images will live e.g. "gs://foo/bar"
+      data_source: "gs://zenml-internal-artifact-store/cv_ship_detection"
```
Comment (strickvl) [inline end-to-end-computer-vision/configs/ingest_data.yaml:8]:
```suggestion
      data_source: "gs://zenml-internal-artifact-store/cv_ship_detection"
```

Was this change needed / intentional?

File: end-to-end-computer-vision/steps/train_model.py [hunk 11]
```diff
@@ -57,6 +58,7 @@ def train_model(
         dataset: Dataset to train the model on.
         data_source: Source where the data lives
         is_quad_gpu_env: Whether we are in an env with 4 gpus
+        is_apple_silicon_env: In case we are running on apple compute
```
Comment (strickvl) [inline end-to-end-computer-vision/steps/train_model.py:61]:
```suggestion
        is_apple_silicon_env: In case we are running on Apple compute
```

**Review by strickvl — CHANGES_REQUESTED** [review]

File: end-to-end-computer-vision/configs/ingest_data.yaml (hunk 10 above)
Comment (AlexejPenner) [reply end-to-end-computer-vision/configs/ingest_data.yaml:8]: good catch

File: end-to-end-computer-vision/README.md (hunk 5 above)
Comment (AlexejPenner) [reply end-to-end-computer-vision/README.md]: good point

File: end-to-end-computer-vision/README.md [hunk 12]
```diff
@@ -26,14 +26,13 @@ things that you'll need to do.
...
-deployed instance of zenml:
+deployed instance of ZenML:
 
 ### Set up your environment
 
 ```bash
 pip install -r requirements.txt
-zenml integration install torch gcp mlflow -y
-pip install label-studio
+zenml integration install torch gcp mlflow label-studio -y
```
Comment (strickvl) [inline end-to-end-computer-vision/README.md:35]:
```suggestion
zenml integration install torch gcp mlflow label_studio -y
```

ZenML integrations have underscores, not hyphens.

File: end-to-end-computer-vision/configs/ingest_data.yaml [hunk 13]
```diff
@@ -5,7 +5,7 @@ steps:
     enable_step_logs: False
     parameters:
       dataset: "datadrivenscience/ship-detection"
-      data_source: "gs://zenml-internal-artifact-store/cv_ship_detection"
+      data_source: <INSERT_HERE>  # Replace this with our path to the data source
```
Comment (strickvl) [inline end-to-end-computer-vision/configs/ingest_data.yaml:8]:
```suggestion
      data_source: <INSERT_HERE>  # Replace this with the path to a data source
```

**Review by strickvl — CHANGES_REQUESTED** [review]

**Review by strickvl — APPROVED** [review]
Let's go!
